tests/trainer/logging_/test_eval_loop_logging.py

```diff
@@ -616,3 +617,17 @@ def validation_step(self, batch, batch_idx):
     )
 
     trainer.fit(model)
+
+
+@RunIf(min_gpus=1)
+def test_move_metrics_to_cpu(tmpdir):
+
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        fast_dev_run=True,
+        amp_backend="native",
+        precision=16,
+        move_metrics_to_cpu=True,
+        gpus=1,
+    )
+    trainer.fit(BoringModel())
```

Comment:
can you assert something? perhaps the result.minimize device in `on_before_backward`

Reply:
Done !